src/go/rdctl/pkg/factoryreset/delete_data.go

```diff
@@ -30,27 +30,38 @@ import (
 
 	dockerconfig "github.com/docker/docker/cli/config"
 	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/directories"
-	p "github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/paths"
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/paths"
 	"github.com/sirupsen/logrus"
 )
 
-func addAppHomeWithoutSnapshots(appHome string) []string {
-	haveSnapshots := false
-	if snapshots, err := os.ReadDir(filepath.Join(appHome, "snapshots")); err == nil {
-		haveSnapshots = len(snapshots) > 0
+func appHomeDirectories(appPaths paths.Paths) []string {
+	// These appHome subdirectories should not be deleted unless they are empty.
+	// The logic below will actually keep all of them as long as at least one is non-empty.
+	excludeDir := map[string]bool{
+		appPaths.Snapshots:       true,
+		appPaths.ContainerdShims: true,
 	}
-	if !haveSnapshots {
-		return []string{appHome}
+	haveExclusions := false
+	for dirname := range excludeDir {
+		files, err := os.ReadDir(dirname)
+		if err == nil && len(files) > 0 {
+			haveExclusions = true
+			break
+		}
+	}
+	if !haveExclusions {
+		return []string{appPaths.AppHome}
 	}
-	appHomeMembers, err := os.ReadDir(appHome)
+	appHomeFiles, err := os.ReadDir(appPaths.AppHome)
```

Comment:
Maybe `appHomeChildren` instead of `appHomeFiles` (since they may be non-regular-files)?

Reply:
I don't like the terms `children` or `members` for directory entries; the correct terminology is `files`. The `stat()` function will tell you what kind of file it is. Theoretically I could see `appHomeDirEntries`, but I find `Files` clearer and just as correct.